### DISCUSSION TITLE
Can I track a connection with hyper?

### DISCUSSION THREAD

**Discussion by hytgbn** [hyperium/hyper#2735] [Q&A]
Hi all,

I'm trying to manage some data over across multiple requests over a connection. This could include some sort of metrics, tcpinfo, or some custom maybe, etc. 

Request Handler needs to be able to access those data and I also want to handle disconnection event so i can log the data.

Is this easily doable with Hyper?

I looked into a possibility to provide custom Acceptor by externding AddrIncoming / AddrStream. However, both of them are private struct, so it'll be more likely copying entire code and manage by myself, which I'd like to avoid is possible.

**Comment by seanmonstar** [comment]
Yea you can track the connection. You don't need to use `Accept`/`Server`, if you want to manage the connections yourself I'd recommend sticking to `hyper::server::conn::Http`.

**Comment by mingmwang** [comment]
I have a similar use case that I need to monitor and react to connection closed, how I can achieve this in hyper ?
